pkg/querytracker/querytracker.go

```diff
@@ -215,6 +215,27 @@ type colUsage struct {
 	usage int
 }
 
+func PopulateColUsageFromMeasure(measureAgg *structs.MeasureAggregator, measureInfoUsage map[string]bool) {
+	if measureAgg == nil {
+		return
+	}
+
+	if measureAgg.ValueColRequest != nil {
+		fields := measureAgg.ValueColRequest.GetFields()
+		for _, field := range fields {
+			measureInfoUsage[field] = true
+		}
+	} else {
+		measureInfoUsage[measureAgg.MeasureCol] = true
+	}
```

Comment:
Is the section in this `if` actually useful to us?
```
		fields := measureAgg.ValueColRequest.GetFields()
		for _, field := range fields {
			measureInfoUsage[field] = true
		}
```
Or should we just skip this `measureAgg` when `ValueColRequest != nil`?

I think this information is only used for `initStarTreeCols()` (which calls `GetTopPersistentAggs()` to get the info), and I think if we have a stats eval query we won't be able to use this info to answer the query anyways, but I might be wrong. Is there a way we can use this for the query, or some other use case where this would be useful?

cc: @nkunal

Reply:
addressed in the latest commit.